app/js/app.js

```diff
@@ -18,46 +18,51 @@
 
 /*jslint node: true */
 /*global angular*/
+(function () {
+    'use strict';
 
-'use strict';
-
-/* App Module */
-
-var app = angular.module('app', [
-    'ngRoute',
-    'ui.dashboard',
-    'app.controllers',
-    'app.datamodels',
-    'app.widgets',
-    'app.directives',
-    'app.services',
-    'app.factories',
-    'app.filters',
-    'vector.config',
-    'nvd3ChartDirectives',
-    'angular-flash.service',
-    'angular-flash.flash-alert-directive'
-]);
-
-app.config(['$routeProvider', function ($routeProvider) {
-    $routeProvider
-        .when('/', {
-            templateUrl: 'partials/dashboard.html',
-            controller: 'DashboardCtrl',
-            title: 'Dashboard - Vector',
-            reloadOnSearch: false
-        })
-        .when('/empty', {
-            templateUrl: 'partials/dashboard.html',
-            controller: 'DashboardCtrl',
-            title: 'Dashboard - Vector',
-            reloadOnSearch: false
-        })
-        .otherwise({
-            redirectTo: '/'
-        });
-}]);
-
-app.config(['flashProvider', function (flashProvider) {
-    flashProvider.errorClassnames.push('alert-danger');
-}]);
+    /**
+    * @name flashConfig
+    * @desc Configures flashProvider to use Bootstrap 3 CSS class.
+    */
+    function flashConfig(flashProvider)  {
+        flashProvider.errorClassnames.push('alert-danger');
+    }
+
+    angular
+        .module('app.routes', ['ngRoute']);
+
+    angular
```

Comment:
Separate these to individual files; consider feature based modules instead of programming-concept based modules; config feature-based modules instead of globally scoping a config module.